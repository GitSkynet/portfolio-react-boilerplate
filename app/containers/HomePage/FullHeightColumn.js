import styled from 'styled-components';
import { Column } from 'hedron';

const FullHeightColumn = styled(Column)`
  height: 100%;
`;

export default FullHeightColumn;
