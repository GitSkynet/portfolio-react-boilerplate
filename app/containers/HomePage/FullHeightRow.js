import styled from 'styled-components';
import { Row } from 'hedron';

const FullHeightRow = styled(Row)`
  height: 100%;
`;

export default FullHeightRow;
