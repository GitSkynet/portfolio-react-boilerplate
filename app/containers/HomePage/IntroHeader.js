import styled from 'styled-components';

const IntroHeader = styled.header`
  height: 100%;
  width: 100vmin;
  z-index: 9;
`;

export default IntroHeader;
