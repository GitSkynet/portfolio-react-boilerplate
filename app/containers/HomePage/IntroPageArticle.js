import styled from 'styled-components';
import { rem } from 'polished';

const IntroPageArticle = styled.article`
  display: flex;
  flex-wrap: wrap;
  position: relative;
  width: 100%;

  @media (min-height: ${rem('424px')}) {
    align-items: center;
  }
`;

export default IntroPageArticle;
