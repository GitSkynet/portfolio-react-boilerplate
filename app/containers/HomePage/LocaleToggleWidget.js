import styled from 'styled-components';
import { sizes } from '../../styles/templates/mediaQueries';

const LocaleToggleWidget = styled.div`
  display: flex;
  justify-content: center;
  position: absolute;
  right: unset;
  top: 1rem;
  width: 100%;
  
  @media (max-height: 400px) and (max-width: ${sizes.m}px) {
    justify-content: flex-end;
    right: 1rem;
  }
`;

export default LocaleToggleWidget;
