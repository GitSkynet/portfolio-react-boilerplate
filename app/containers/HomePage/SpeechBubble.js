import styled from 'styled-components';
import { rem } from 'polished';

import mq from '../../styles/templates/mediaQueries';

const SpeechBubble = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  
  ${mq.m`
    padding-right: ${rem('114px')}; // Keep room for site navigation
  `}
`;

export default SpeechBubble;
