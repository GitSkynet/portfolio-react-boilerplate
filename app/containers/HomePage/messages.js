/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'portfolio.page.index.title',
    defaultMessage: 'Introduction',
  },
  greeting: {
    id: 'portfolio.page.index.header.greeting',
    defaultMessage: 'WELCOME',
  },
  intro: {
    id: 'portfolio.page.index.header.intro',
    defaultMessage: 'I am {name}',
  },
  trade: {
    id: 'portfolio.page.index.header.trade',
    defaultMessage: 'FRONT-END DEVELOPER',
  },
});
