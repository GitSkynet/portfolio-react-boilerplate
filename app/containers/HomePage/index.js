/*
 * HomePage
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Helmet } from 'react-helmet';

import messages from './messages';

import IntroHeader from './IntroHeader';
import IntroPageArticle from './IntroPageArticle';
import SpeechBubble from './SpeechBubble';
import MyPicture from './itsme-trevi-rome.jpg';
import FlexRow from '../../components/FlexRow';
import FlexColumn from '../../components/FlexColumn';

class HomePage extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  componentDidMount() {
    this.animateSvg();
  }

  componentDidUpdate() {
    this.animateSvg();
  }

  animateSvg() {
    if (!window) return;

    import('snapsvg')
      .then((Snap) => {
        const SVG_EL = Snap(this.svgDiv);
        const TIMING = window.mina;
        const ANIMATION_DURATION = 200; // milliseconds
        const ANIMATION_DELAY = 200; // milliseconds

        const INFOPAGE_BALLOON_ANIMATION_TIMELINE = [
          {
            elementId: 'p1',
            animationAttributes: {
              d: 'M0 439.8v140.3l78.7 153.1H159L0 439.8',
            },
            animationDuration: ANIMATION_DURATION / 2,
            animationDelay: 1,
            animationTimingFunction: TIMING.linear,
            dFrom: 'M0 439.8v140.3l1 2h76.1z',
            dTo: '',
          },
          {
            elementId: 'p2',
            animationAttributes: {
              d: 'M128.3 790.4l30.7-57.2H78.7l-26.2 57.2z',
            },
            animationDelay: ANIMATION_DELAY / 2,
            animationDuration: ANIMATION_DURATION / 2,
            animationTimingFunction: TIMING.linear,
            dFrom: 'M159,733.2 v0 H78.700002 78.7z',
            dTo: '',
          },
          {
            elementId: 'p3',
            animationAttributes: {
              d: 'M5.3 702.1l47.2 88.3h75.8l-49.1-88.3z',
            },
            animationDelay: ANIMATION_DELAY,
            animationDuration: ANIMATION_DURATION,
            animationTimingFunction: TIMING.linear,
            dFrom: 'M52.5,790.4 v0 h75.8 v0z',
            dTo: '',
          },
          {
            elementId: 'p4',
            animationAttributes: {
              d: 'M5.3 702.1l180.1-335.9h70.2L79.2 702.1z',
            },
            animationDelay: ANIMATION_DELAY,
            animationDuration: ANIMATION_DURATION,
            animationTimingFunction: TIMING.linear,
            dFrom: 'M5.3,702.1 v0 h73.9 v0z',
            dTo: '',
          },
          {
            elementId: 'p5',
            animationAttributes: {
              d: 'M595.3 373.2c0-4.4-3.6-8.1-8.1-8.1H185.91L40.836 635.841 587.3 635.8c4.4 0 8.1-3.6 8.1-8.1V373.2z',
              opacity: 1,
            },
            animationDelay: ANIMATION_DELAY,
            animationDuration: ANIMATION_DURATION,
            animationTimingFunction: TIMING.linear,
            dFrom: 'M185.91 365.1L0 0l-145.073 270.741z',
            dTo: '',
          },
          {
            elementId: 'g2',
            animationAttributes: {
              opacity: 1,
            },
            animationDelay: ANIMATION_DELAY,
            animationDuration: ANIMATION_DURATION,
            animationTimingFunction: TIMING.linear,
          },
          {
            elementId: 't3',
            animationAttributes: {
              opacity: 1,
            },
            animationDelay: ANIMATION_DELAY,
            animationDuration: 1,
            animationTimingFunction: TIMING.linear,
          },
          {
            elementId: 't3',
            animationAttributes: {
              'font-size': '40px',
            },
            animationDelay: ANIMATION_DELAY,
            animationDuration: ANIMATION_DURATION,
            animationTimingFunction: TIMING.linear,
          },
          {
            elementId: 't3',
            animationAttributes: {
              'font-size': '20px',
            },
            animationDelay: ANIMATION_DELAY,
            animationDuration: ANIMATION_DURATION,
            animationTimingFunction: TIMING.bounce,
          },
        ];

        const START_FRAME = 0; // 0 is the first frame

        let animationTimer = setTimeout(
          function animate(_timeline, currentFrameIndex) {
            const currentFrame = _timeline[currentFrameIndex];
            const element = SVG_EL.select(`#${currentFrame.elementId}`);

            element.animate(
              currentFrame.animationAttributes,
              currentFrame.animationDuration,
              currentFrame.animationTimingFunction,
            );

            if (currentFrameIndex < INFOPAGE_BALLOON_ANIMATION_TIMELINE.length - 1) {
              /* eslint no-param-reassign: "off" */
              animationTimer = setTimeout(
                animate,
                _timeline[currentFrameIndex + 1].animationDelay,
                INFOPAGE_BALLOON_ANIMATION_TIMELINE,
                (currentFrameIndex += 1),
              );
            } else {
              clearTimeout(animationTimer);
            }
          },
          INFOPAGE_BALLOON_ANIMATION_TIMELINE[START_FRAME].animationDelay,
          INFOPAGE_BALLOON_ANIMATION_TIMELINE,
          START_FRAME,
        );
      });
  }

  render() {
    return (
      <IntroPageArticle>
        <Helmet>
          <title>Welkom op Johan Meester zijn portfolio</title>
          <meta name="description" content="Intro pagina van Johan Meester zijn portfolio" />
        </Helmet>
        <FlexRow fullHeight>
          <FlexColumn fullHeight fluid>
            <IntroHeader>
              <h2 hidden aria-hidden="false"><FormattedMessage {...messages.title} /></h2>
              <SpeechBubble
                innerRef={
                  (d) => {
                    this.svgDiv = d;
                  }
                }
              >
                <svg
                  width="100%"
                  height="100%"
                  viewBox="0 365.1 595.3 425.3"
                  preserveAspectRatio="xMinYMid"
                >
                  <g id="g1">
                    <path
                      id="p1"
                      fill="#212120"
                      d="M-754.1,133.2v140.3l78.7,153.1h80.3L-754.1,133.2z"
                    />
                    <path id="p2" fill="#818180" d="M159,733.2 v0 H78.700002 78.7z" />
                    <path id="p3" fill="#414140" d="M52.5,790.4 v0 h75.8 v0z" />
                    <path id="p4" fill="#818180" d="M5.3,702.1 v0 h73.9 v0z" />
                    <path
                      id="p5"
                      fill="#575756"
                      d="M-31.8,309.6c0-4.4-3.6-8.1-8.1-8.1h-401.2l-145.1,270.6h546.3 c4.4,0,8.1-3.6,8.1-8.1V309.6L-31.8,309.6z"
                      style={{ opacity: 0 }}
                    />
                  </g>
                  <path d="M172 440h387.2v141.4h-387z" fill="none" />
                  <FormattedMessage {...messages.greeting} >
                    {
                      (message) => (
                        <text
                          fill="#575756"
                          fontSize="65.685"
                          x="175"
                          y="500"
                        >{message}</text>
                      )
                    }
                  </FormattedMessage>
                  <FormattedMessage {...messages.greeting} >
                    {
                      (message) => (
                        <text
                          fill="#95C11F"
                          fontSize="65.685"
                          x="172"
                          y="496"
                        >{message}</text>
                      )
                    }
                  </FormattedMessage>
                  <g id="g2" style={{ opacity: 0 }}>
                    <FormattedMessage {...messages.intro} values={{ name: 'Johan Meester' }} >
                      {
                        (message) => (
                          <text
                            id="t2"
                            fill="#FFF"
                            fontSize="47"
                            textAnchor="end"
                            x="571"
                            y="545"
                          >{message}</text>
                        )
                      }
                    </FormattedMessage>
                    <defs>
                      <mask id="m1">
                        <circle cx="553" cy="407" fill="#fff" r="32.5" />
                      </mask>
                    </defs>
                    <image
                      xmlnsXlink="http://www.w3.org/1999/xlink"
                      xlinkHref={MyPicture}
                      preserveAspectRatio="none"
                      x="520"
                      y="374"
                      width="65"
                      height="65"
                      mask="url('#m1')"
                    />
                  </g>
                  <FormattedMessage {...messages.trade} >
                    {
                      (message) => (
                        <text
                          id="t3"
                          style={{ fontSize: '20px', textAnchor: 'end', opacity: 0 }}
                          fill="#FFF"
                          x="571"
                          y="580"
                        >{message}</text>
                      )
                    }
                  </FormattedMessage>
                </svg>
              </SpeechBubble>
            </IntroHeader>
          </FlexColumn>
        </FlexRow>
      </IntroPageArticle>
    );
  }
}

export default HomePage;
