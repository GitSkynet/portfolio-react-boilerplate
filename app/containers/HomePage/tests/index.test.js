import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HomePage from '../index';

describe('<HomePage />', () => {
  it('should render the page message', () => {
    const renderedComponent = shallow(
      <HomePage />
    );
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
