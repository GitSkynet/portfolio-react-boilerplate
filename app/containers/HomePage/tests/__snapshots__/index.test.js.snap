// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render the page message 1`] = `
<IntroPageArticle__IntroPageArticle>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Welkom op Johan Meester zijn portfolio
    </title>
    <meta
      content="Intro pagina van Johan Meester zijn portfolio"
      name="description"
    />
  </HelmetWrapper>
  <FlexRow__FlexRow
    fullHeight={true}
  >
    <FlexColumn__FlexColumn
      fluid={true}
      fullHeight={true}
    >
      <IntroHeader__IntroHeader>
        <h2
          aria-hidden="false"
          hidden={true}
        >
          <FormattedMessage
            defaultMessage="Introduction"
            id="portfolio.page.index.title"
            values={Object {}}
          />
        </h2>
        <SpeechBubble__SpeechBubble
          innerRef={[Function]}
        >
          <svg
            height="100%"
            preserveAspectRatio="xMinYMid"
            viewBox="0 365.1 595.3 425.3"
            width="100%"
          >
            <g
              id="g1"
            >
              <path
                d="M-754.1,133.2v140.3l78.7,153.1h80.3L-754.1,133.2z"
                fill="#212120"
                id="p1"
              />
              <path
                d="M159,733.2 v0 H78.700002 78.7z"
                fill="#818180"
                id="p2"
              />
              <path
                d="M52.5,790.4 v0 h75.8 v0z"
                fill="#414140"
                id="p3"
              />
              <path
                d="M5.3,702.1 v0 h73.9 v0z"
                fill="#818180"
                id="p4"
              />
              <path
                d="M-31.8,309.6c0-4.4-3.6-8.1-8.1-8.1h-401.2l-145.1,270.6h546.3 c4.4,0,8.1-3.6,8.1-8.1V309.6L-31.8,309.6z"
                fill="#575756"
                id="p5"
                style={
                  Object {
                    "opacity": 0,
                  }
                }
              />
            </g>
            <path
              d="M172 440h387.2v141.4h-387z"
              fill="none"
            />
            <FormattedMessage
              defaultMessage="WELCOME"
              id="portfolio.page.index.header.greeting"
              values={Object {}}
            />
            <FormattedMessage
              defaultMessage="WELCOME"
              id="portfolio.page.index.header.greeting"
              values={Object {}}
            />
            <g
              id="g2"
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <FormattedMessage
                defaultMessage="I am {name}"
                id="portfolio.page.index.header.intro"
                values={
                  Object {
                    "name": "Johan Meester",
                  }
                }
              />
              <defs>
                <mask
                  id="m1"
                >
                  <circle
                    cx="553"
                    cy="407"
                    fill="#fff"
                    r="32.5"
                  />
                </mask>
              </defs>
              <image
                height="65"
                mask="url('#m1')"
                preserveAspectRatio="none"
                width="65"
                x="520"
                xlinkHref="IMAGE_MOCK"
                xmlnsXlink="http://www.w3.org/1999/xlink"
                y="374"
              />
            </g>
            <FormattedMessage
              defaultMessage="FRONT-END DEVELOPER"
              id="portfolio.page.index.header.trade"
              values={Object {}}
            />
          </svg>
        </SpeechBubble__SpeechBubble>
      </IntroHeader__IntroHeader>
    </FlexColumn__FlexColumn>
  </FlexRow__FlexRow>
</IntroPageArticle__IntroPageArticle>
`;
